src/pointer-analysis/value_set_dereference.cpp

```diff
@@ -147,8 +146,9 @@ exprt value_set_dereferencet::dereference(const exprt &pointer)
   std::cout << '}' << std::endl;
 #endif
 
-  for(const auto &value : retained_values)
-    values.push_back(build_reference_to(value, compare_against_pointer, ns));
+  std::list<valuet> values =
+    make_range(retained_values).map([&](const exprt &value){
+      return build_reference_to(value, compare_against_pointer, ns);});
```

Comment:
Is this better than `std::transform(value.begin(), value.end(), std::back_inserter(retained_values), [&](const exprt &value){ return build_reference_to(value, compare_against_pointer, ns);} );`?

Reply:
It's mostly the same thing as `std::transform` but I find the range version clearer